### DISCUSSION TITLE
Uso de Using e Dispose

### DISCUSSION THREAD

**Discussion by RonaldoBueno16** [balta-io/forum#326] [Tira Dúvidas] [answered]
Professor, boa noite!

Gostaria de tirar uma dúvida:

Sempre que eu instanciar um objeto que não seja uma conexão de banco de dados, eu devo implementar o Dispose para remove-lo da memória?

Pois existe o Garbage Collector que faz o gerenciamento de variaveis que não estão sendo utilizadas e é deletada automaticamente, mas isso é executado o tempo todo? Pesa a aplicação caso eu não delete um objeto "pesado"?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @RonaldoBueno16, como vai?

É uma ótima questão... O GC tem algumas fases que varrem a Heap em busca de objetos que não estão mais em uso, normalmente, objetos dentro do Stack Frame que acabou de ser desalocado, já são marcados para serem removidos, caso não estejam em um ambiente global ou sendo utilizados por outros frames/objetos.

Resumindo, quando uma função termina, o stack frame dela já é inteligente para marcar os objetos a serem removidos da memória.

Neste caso que você citou, o Dispose está mais relacionado a encerrar a conexão do que ao tempo de vida do objeto. A preocupação não é nem o objeto existir na memória, mas sim fechar a conexão o quão antes possível.

Até onde sei o Dispose ajuda nessa captura mas não garante que o objeto vai ser removido na Gen 0 do GC, ainda tem outros fatores que podem influenciar nisso.
